handlers/api.go

```diff
@@ -3048,7 +3048,7 @@ func RegisterEthpoolSubscription(w http.ResponseWriter, r *http.Request) {
 
 	subscriptionCount, err := db.GetAppSubscriptionCount(claims.UserID)
 	if err != nil || subscriptionCount >= 5 {
-		w.WriteHeader(http.StatusInternalServerError)
+		w.WriteHeader(http.StatusConflict)
 		sendErrorResponse(w, r.URL.String(), "reached max subscription limit")
```

Comment:
**issue:** This combination doesn't seem to make sense.
First we set the header to **StatusConflict** even if we have a normal error.
Then we set the header to **StatusBadRequest** in `sendErrorResponse`.

Reply:
**question:** The new changes alter the way we fill `w`.
Before it was a
```
j := json.NewEncoder(w)
response := &types.ApiResponse{}
response.Status = errMsg
j.Encode(response)
```
now it is a
`fmt.Fprintln(w, errorMsg)`

Is such a change allowed?

Reply:
even nicer catch, fixed

Reply:
It was fixed for one of two cases, please also fix the other, just search for USER_SUBSCRIPTION_LIMIT in api.go.